File: 24_ReverseList/ReverseList.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    # 返回ListNode
    def ReverseList(self, pHead):
        # write code here
        reversedHead = None
        p = pHead
        pre = None
        while p:
            q = p.next
            if q is None:
                reversedHead = p
            p.next = pre
            pre = p
            p = q
        return reversedHead

File: 24_ReverseList/ReverseList.cpp
#include <iostream>
#include <vector>
#include <string>
#include <deque>
#include <set>
#include <map>
#include <queue>
#include <stack>
#include <algorithm>
#include <cstring>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution
{
  public:
    ListNode *ReverseList(ListNode *pHead)
    {
        ListNode *reversedHead = nullptr;
        ListNode *p = pHead;
        ListNode *pre = nullptr;
        while (p)
        {
            ListNode *q = p->next;
            if (q == nullptr)
                reversedHead = p;
            p->next = pre;
            pre = p;
            p = q;
        }
        return reversedHead;
    }
};